### ISSUE TITLE
reproduce？

### ISSUE THREAD

**Issue by ccccccccxy** [Hamidreza-Hashempoor/FastTracker#17]
How can I reproduce your experimental results? Are all the required parameters for each dataset (e.g., track_thresh, match_thresh, etc.) already reflected in the JSON files under the configsfolder and in lines 150-170 of mot_evaluator.py?

**Comment by Hamidreza-Hashempoor** [comment]
No,

To reproduce the best performance reported in our paper and in MOT Challenge server, you need to tune the hyperparameters for **each video sequence individually** in [lines 150-170 mot_evaluator.py](https://github.com/Hamidreza-Hashempoor/FastTracker/blob/main/yolox/evaluators/mot_evaluator.py). This is done by editing the corresponding JSON config file in ./configs/ with sequence-specific values*.


Even with these differences the performance would not deviate more than 3-4 percent in the metrics at the end.

**Comment by MohamedNagyMostafa** [comment]
What about the evaluation on the MOT17 test split? I tried to run the code as its and submit the result on the MOTChallenge website, it gives about ~4% percent less in HOTA compared to the results stated in Table VIII.

**Comment by ccccccccxy** [comment]
The results I ran on the MOT17 training set (not the test set) are shown below. The MOTA is only 74 and the HOTA is only 50, and I am not very sure about the reasons for this. The results are attached at the end.
HOTA: FastTracker-pedestrian       HOTA      DetA      AssA      DetRe     DetPr     AssRe     AssPr     LocA      OWTA      HOTA(0)   LocA(0)   HOTALocA(0)
MOT17-02-FRCNN                     24.249    46.628    12.808    71.326    53.641    14.594    61.54     83.622    30.124    31.308    73.161    22.905    
MOT17-04-FRCNN                     70.838    83.047    60.527    90.289    88.125    63.066    89.424    91.506    73.918    78.742    89.097    70.157    
MOT17-05-FRCNN                     41.558    50.44     34.685    73.246    57.816    40.554    67.55     84.651    50.254    52.508    76.222    40.022    
MOT17-09-FRCNN                     45.781    62.306    33.894    79.252    69.686    39.055    66.816    86.274    51.777    55.704    80.174    44.66     
MOT17-10-FRCNN                     25.356    47.255    14.23     65.9      56.492    15.321    63.982    80.921    30.212    32.668    70.889    23.158    
MOT17-11-FRCNN                     37.831    58.292    24.703    76.888    66.415    27.088    73.296    85.971    43.552    47.956    77.935    37.375    
MOT17-13-FRCNN                     43.866    56.416    34.428    72.539    65.012    38.871    65.303    82.963    49.908    55.39     76.343    42.287    
COMBINED                           50.583    62.656    41.461    79.824    70.504    44.382    77.799    87.313    57.346    59.219    81.08     48.015    

CLEAR: FastTracker-pedestrian      MOTA      MOTP      MODA      CLR_Re    CLR_Pr    MTR       PTR       MLR       sMOTA     CLR_TP    CLR_FN    CLR_FP    IDSW      MT        PT        ML        Frag      
MOT17-02-FRCNN                     50.708    64.813    57.128    95.049    71.482    93.548    1.6129    4.8387    17.263    17661     920       7046      1193      58        1         3         263       
MOT17-04-FRCNN                     94.018    87.045    95.185    98.82     96.452    100       0         0         81.215    46996     561       1729      555       83        0         0         242       
MOT17-05-FRCNN                     54.793    71.935    60.012    93.35     73.685    78.947    18.797    2.2556    28.594    6457      460       2306      361       105       25        3         108       
MOT17-09-FRCNN                     75.493    75.202    78.911    96.319    84.693    96.154    3.8462    0         51.608    5129      196       927       182       25        1         0         42        
MOT17-10-FRCNN                     55.573    65.282    63.424    90.038    77.185    89.474    10.526    0         24.313    11560     1279      3417      1008      51        6         0         440       
MOT17-11-FRCNN                     71.905    72.098    76.388    96.079    82.992    90.667    6.6667    2.6667    45.098    9066      370       1858      423       68        5         2         157       
MOT17-13-FRCNN                     67.6      71.95     73.493    92.536    82.933    86.364    10.909    2.7273    41.644    10773     869       2217      686       95        12        3         232       
COMBINED                           74.565    76.82     78.49     95.855    84.663    88.828    9.1575    2.0147    52.345    107642    4655      19500     4408      485       50        11        1484      

Identity: FastTracker-pedestrian   IDF1      IDR       IDP       IDTP      IDFN      IDFP      
MOT17-02-FRCNN                     24.607    28.664    21.557    5326      13255     19381     
MOT17-04-FRCNN                     74.46     75.375    73.568    35846     11711     12879     
MOT17-05-FRCNN                     48.444    54.908    43.341    3798      3119      4965      
MOT17-09-FRCNN                     56.867    60.77     53.435    3236      2089      2820      
MOT17-10-FRCNN                     21.678    23.483    20.131    3015      9824      11962     
MOT17-11-FRCNN                     39.106    42.189    36.443    3981      5455      6943      
MOT17-13-FRCNN                     49.87     52.757    47.283    6142      5500      6848      
COMBINED                           51.24     54.627    48.248    61344     50953     65798     

Count: FastTracker-pedestrian      Dets      GT_Dets   IDs       GT_IDs    
MOT17-02-FRCNN                     24707     18581     768       62        
MOT17-04-FRCNN                     48725     47557     528       83        
MOT17-05-FRCNN                     8763      6917      363       133       
MOT17-09-FRCNN                     6056      5325      131       26        
MOT17-10-FRCNN                     14977     12839     754       57        
MOT17-11-FRCNN                     10924     9436      406       75        
MOT17-13-FRCNN                     12990     11642     552       110       
COMBINED                           127142    112297    3502      546       

Timing analysis:
MotChallenge2DBox.get_raw_seq_data                                     0.8232 sec
MotChallenge2DBox.get_preprocessed_seq_data                            0.4996 sec
HOTA.eval_sequence                                                     0.7748 sec
CLEAR.eval_sequence                                                    0.1254 sec
Identity.eval_sequence                                                 0.1118 sec
Count.eval_sequence                                                    0.0000 sec
eval_sequence                                                          2.3470 sec
Evaluator.evaluate                                                     2.7419 sec
Are these metrics reasonable on the training set, or did I make some mistake that caused the performance to drop?

**Comment by MohamedNagyMostafa** [comment]
Yes same here. This is after the update they did recently. I had the older version which was giving about 69% HOTA on training. The thing is that even the old version does not achieve the same result on the test dataset!

The author mentioned about individual finetuning for streams, which im not sure why they do not provide the params since they have the results?!!. But the same problem happens on testing as well makes it suspicious. 

The purpose of adding code of a publication is to validate the reported results, and it has to be close to match it.

**Comment by ccccccccxy** [comment]
The author of the paper stated that his detector underwent intensive training, and he used fine-tuned parameters, specifically tailored for the sequence, which is why the metrics are better. The author claimed that if you use your own (model or data), the gap (in performance) wouldn't exceed 3% to 4%. Mohamed Nagy Mostafa, you mentioned that you have an old version that can achieve a HOTA score of 69. Was this result obtained on the training set or the test set? Can your old model achieve a HOTA score of 69 on the training set?

**Comment by MohamedNagyMostafa** [comment]
Yeah, I noticed that one of the authors wrote about the detector's intensive training, which neglects the contribution of the work. One of the critical points in object tracking research is to use the same detector to demonstrate that the contribution comes from the proposed method. If a method outperforms due to extra training for a detector, then the contribution of the proposed work is negligible, because using the extra-trained detector on other benchmarks will also improve their performances and outperform the proposed method. Based on my own experiment on this code in Github, it does not outperform the benchmarks (Given the same detections), which contradicts the paper!

Regarding HOTA, yes it achieves 69% in HOTA, but I am using the official TrackEval from MOTChallenge, not the one provide in this repository.

**Comment by Hamidreza-Hashempoor** [comment]
Please note that the main goal of FastTracker is vehicle tracking in traffic scenarios, and the parameters are tuned for best performance on this target.

To get the best results of pedestrian targets (e.g. MOT17, MOT20, etc.), should you go ahead and tune config for each individual file.
For the exact details (such as used configs in the exps, or detector check points, etc.) should you contact the co-author of this work to provide these info as they are a private organization.*
While noting again, the results would not deviate more than 3-4 percents after tuning, as I recall.
 
(*PS: I leaved that organization in this moment)

**Comment by MohamedNagyMostafa** [comment]
I have multiple comments regarding that. In terms of publication ethicality, the code has to provide the exact results mentioned in the paper; otherwise, there is no prove that numbers in the paper are actually obtained.

Second, tuning every stream individually makes the framework not generalized! In fact, it disprove the calm of online tracking framework if the params change based on scenario!! Specially, this has not been mentioned in the paper.
